client/my-sites/domains/domain-management/list/free-domain-item.jsx

```diff
@@ -28,6 +26,27 @@ export default function FreeDomainItem( {
 		event.stopPropagation();
 		onMakePrimary( domain.domain );
 	};
+	const [ isDialogVisible, setDialogVisible ] = useState( false );
+	const [ openDialog, closeDialog ] = useMemo( () => {
```

Comment:
Out of curiosity... is there any advantage here to use `useMemo` for `openDialog`/`closeDialog` function, instead of simply passing` ()=> setDialogVisible(true)` to handler props?

Reply:
> Out of curiosity... is there any advantage here to using `useMemo` for `openDialog`/`closeDialog` functions, instead of simply passing` ()=> setDialogVisible(true)` to handler props?

Not that I can think of - to be honest, this is just some early optimization and overkill 😆 
I'll remove it.